src/org/labkey/remoteapi/plate/PlateSetParams.java

```diff
@@ -1,122 +1,135 @@
 package org.labkey.remoteapi.plate;
 
-
 import org.json.JSONObject;
 
 public class PlateSetParams
 {
-    private String _name;
+    private boolean _archived;
+    private String _containerId;
+    private String _containerName;
+    private String _containerPath;
+    private String _created;
+    private Integer _createdBy;
     private String _description;
-    private PlateSetType _type;
+    private String _modified;
+    private Integer _modifiedBy;
+    private String _name;
+    private Integer _plateCount;
     private String _plateSetId;
-    private Integer _rowId;
+    private Integer _primaryPlateSetId;
     private Integer _rootPlateSetId;
+    private Integer _rowId;
+    private boolean _template;
+    private CreatePlateSetParams.PlateSetType _plateSetType;
 
-    public PlateSetParams()
+    public PlateSetParams(JSONObject json)
     {
+        if (json.has("archived"))
+            _archived = json.getBoolean("archived");
+        if (json.has("containerId"))
+            _containerId = json.getString("containerId");
```

Comment:
FYI, you can make this a bit more concise by using `json.opt*` methods to get optional values.
```suggestion
        _archived = json.optBoolean("archived");
        _containerId = json.optString("containerId", null);
        ...etc.
```
One warning though; none of the default values are `null`, so you'll have to specify those defaults explicitly (e.g. `json.optInt("createdBy", null)`)

Reply:
Note that `json.optInt("createdBy", null)` is not an option since this method returns a primitive `int`. `json.optString("containerID", null)` is totally valid and useful.

There are also `optIntegerObject(String, Integer)`, `optBooleanObject(String, Boolean)`, etc. if you're stashing `Integer` or `Boolean` values and want to default to `null`.